spec/chore/queues/sqs/consumer_spec.rb

```diff
@@ -1,145 +1,194 @@
+    # effectively stub the block.
+    def consume(&block)
+      block = Proc.new{} unless block_given?
+      consumer.consume(&block)
     end
 
-    it 'should look up the queue based on the queue url' do
-      expect(queues).to receive(:[]).with(queue_url).and_return(queue)
-      consumer.consume
+    before do
+      allow(Aws::SQS::Client).to receive(:new).and_return(sqs)
+      allow(Aws::SQS::Queue).to receive(:new).and_return(queue)
+      allow(sqs).to receive(:receive_message).and_return(receive_message_result)
+      allow(queue).to receive(:receive_messages).and_return(sqs.receive_message)
+      allow(message).to receive(:attributes).and_return({ 'ApproximateReceiveCount' => rand(10) })
     end
 
-    context "should receive a message from the queue" do
-
-      it 'should use the default size of 10 when no queue_polling_size is specified' do
-        expect(queue).to receive(:receive_messages).with(:limit => 10, :attributes => [:receive_count])
-        consumer.consume
+    describe "consuming messages" do
+      before do
+        allow(consumer).to receive(:running?).and_return(true, false)
       end
 
-      it 'should respect the queue_polling_size when specified' do
-        allow(Chore.config).to receive(:queue_polling_size).and_return(5)
-        expect(queue).to receive(:receive_messages).with(:limit => 5, :attributes => [:receive_count])
-        consumer.consume
+      context "should create objects for interacting with the SQS API" do
+        it 'should create an sqs client' do
+          expect(queue).to receive(:receive_messages)
+          consume
+        end
+
+        it "should only create an sqs client when one doesn't exist" do
+          allow(consumer).to receive(:running?).and_return(true, true, true, true, false, true, true)
+          expect(Aws::SQS::Client).to receive(:new).exactly(:once)
+          consume
+        end
+
+        it 'should look up the queue url based on the queue name' do
+          expect(sqs).to receive(:get_queue_url).with(queue_name: queue_name)
+          consume
+        end
+
+        it 'should create a queue object' do
+          expect(consumer.send(:queue)).to_not be_nil
+          consume
+        end
+
+        # TODO: This seems like it's no longer necessary. If it is, fix and reenable it, otherwise delete
+        xit 'should look up the queue based on the queue url' do
+          expect(sqs).to receive(:get_queue_url).with(:queue_name=>queue_name).and_return(queue_url)
+          expect(queue).to receive(:[]).with(queue_url).and_return(queue)
+          consume
+        end
       end
-    end
 
-    it "should check the uniqueness of the message" do
-      allow_any_instance_of(Chore::DuplicateDetector).to receive(:found_duplicate?).with(message_data).and_return(false)
-      consumer.consume
-    end
+      context "should receive a message from the queue" do
+        before do
+          allow(consumer).to receive(:queue).and_return(queue)
+        end
```

Comment:
Is this required if you've previously stubbed it with `allow(Aws::SQS::Queue).to receive(:new).and_return(queue)`?